src/comments.js

```diff
@@ -154,7 +154,12 @@ function attach(comments, ast, text) {
 
       if (followingNode) {
         // Always a leading comment.
-        addLeadingComment(followingNode, comment);
+        if (followingNode.type === "Identifier" &&
```

Comment:
Can you provide code examples of where this moves the comment?

We want to also move comments in if/else blocks, and I'd like to avoid putting so much logic into this function. It already has to deal with a lot of complex logic about large-scale comment placement.

What if we created a function `determineFollowingNode` that takes the current following node and the enclosing node and returns a new node? That way you could special-case anything and return a new following node to use.

Then use it in `decorateComment`. At the end, change

```js
  if (followingNode) {
    comment.followingNode = followingNode;
  }

```

To this:

```js
  if (followingNode) {
    comment.followingNode = determineFollowingNode(
      followingNode, comment.enclosingNode
    );
  }
```

In `determineFollowingNode` you'd do your special-casing and return whichever node you want.